activejob/lib/active_job/arguments.rb

```diff
@@ -147,7 +147,9 @@ def serialize_hash_key(key)
       end
 
       def transform_symbol_keys(hash, symbol_keys)
-        hash.transform_keys do |key|
+        # NOTE: #transform_keys doesn't play well with
+        # HashWithIndifferentAccess hashes so we call #to_h here.
```

Comment:
Can we be specific about the issue here?

Reply:
Yes, I will rewrite my comment.